Documentation/cmdref/cilium-agent.md

```diff
@@ -83,6 +83,7 @@ cilium-agent
       --sidecar-istio-proxy-image string            Regular expression matching compatible Istio sidecar istio-proxy container image names (default "cilium/istio_proxy")
       --single-cluster-route                        Use a single cluster route instead of per node routes
       --socket-path string                          Sets daemon's socket path to listen for connections (default "/var/run/cilium/cilium.sock")
+      --sockops-enable                              Enable sockops when kernel supported
```

Comment:
To avoid another flag, what are the downsides of enabling it automatically if we autodetect if the kernel supports it?

Reply:
I think we probably should do that once all bits land and were exposed for a bit; then auto-detect could enable it for users (otherwise they might not be aware of it).

Reply:
Agree on waiting a bit to automatically enable it. Also we need to pick a kernel which has a minimal set of features needs (4.18 I believe for this series) and check we have all needed backports.

The other work in-progress is getting CI running consistently with this series on all kernels we will support. Currently we have bpf-next running but that is only running a limited subset of the CI at the moment for reasons unrelated to this series. We would want 4.X CI runs as well before enabling by default IMO because kernel differences can matter.

Finally one TODO still after this series is to get conntrack running fully with this. There are a couple paths here that may bypass conntrack updates. This needs a bit more analysis.

All the above becomes easier if we get the base series in.

Reply:
^^^ I for what its worth I was wrong on one point above the CI is complete on next kernel testing ;) but still relevant if we enable on 4.18 we need to test that as well. Fairly sure it *should* work seeing kernel selftests are passing, etc. but still best to cover all the boxes before flipping the switch.

Reply:
@aanm marking resolved seems there is some consensus around leaving the flag for the time being.

Reply:
:+1: